homeassistant/components/shelly/update.py

```diff
@@ -0,0 +1,229 @@
+"""Support for Synology DSM update platform."""
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any, Final
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+)
+from homeassistant.components.update.const import UpdateEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import BlockDeviceWrapper, RpcDeviceWrapper
+from .const import BLOCK, DATA_CONFIG_ENTRY, DOMAIN
+from .entity import (
+    RestEntityDescription,
+    RpcEntityDescription,
+    ShellyRestAttributeEntity,
+    ShellyRpcAttributeEntity,
+    async_setup_entry_rest,
+    async_setup_entry_rpc,
+)
+from .utils import get_device_entry_gen
+
+
+@dataclass
+class RestUpdateEntityDescription(RestEntityDescription, UpdateEntityDescription):
+    """Class to describe a REST update entity."""
+
+    current_version: Callable[[dict], str | None] | None = None
+    latest_version: Callable[[dict], str | None] | None = None
+    release_url: str | None = None
+    install_callback: Callable[
+        [BlockDeviceWrapper], Coroutine[Any, Any, None]
+    ] | None = None
+
+
+@dataclass
+class RpcUpdateEntityDescription(RpcEntityDescription, UpdateEntityDescription):
+    """Class to describe a RPC update entity."""
+
+    current_version: Callable[[dict], str | None] | None = None
+    latest_version: Callable[[dict[str, dict]], str | None] | None = None
+    release_url: str | None = None
+    install_callback: Callable[
+        [RpcDeviceWrapper], Coroutine[Any, Any, None]
+    ] | None = None
+
+
+BLOCK_UPDATE_ENTITIES: Final = {
+    "fwupdate": RestUpdateEntityDescription(
```

Comment:
This means that we now have a polling based sensor which is now default enabled so polling will occur for all Gen1 devices

Reply:
ahh ... you are right - the binary sensors where also disabled by default ... will change this for the update entities, too.